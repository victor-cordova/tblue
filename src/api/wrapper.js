// @flow
/**
 * The functions in this file assume that you are using the trello API which
 * receives the key and token as query params
 */

import axios from 'axios';
import qs from 'qs';

import env from '~/common/env';

export const getQueryParams = (baseParams: Object): string => {
    const key = env.getTrelloApiKey();
    const token = env.getAppToken();
    const fullParams = {
        ...baseParams,
        key,
        token,
    };
    return qs.stringify(fullParams, { addQueryPrefix: true });
};

export const doPost = (baseParams: Object, getUrl: (queryParams: string) => string) => {
    const queryParams = getQueryParams(baseParams);
    return axios.post(getUrl(queryParams));
};

export const doGet = (baseParams: Object, getUrl: (queryParams: string) => string) => {
    const queryParams = getQueryParams(baseParams);
    return axios.get(getUrl(queryParams));
};