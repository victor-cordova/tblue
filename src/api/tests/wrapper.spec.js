import { getQueryParams } from '../wrapper';

jest.mock('../../common/env', () => ({
    getTrelloApiKey: () => '<TrelloApiKey>',
    getAppToken: () => '<AppToken>',
}));

describe('getQueryParams', () => {
    it('converts params to string correctly', () => {
        const baseOptions = {
            name: "John Francis 213",
            surname: "Doe",
        };
        expect(getQueryParams(baseOptions)).toMatchSnapshot();
    });
});
