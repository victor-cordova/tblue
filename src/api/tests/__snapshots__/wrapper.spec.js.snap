// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQueryParams converts params to string correctly 1`] = `"?name=John%20Francis%20213&surname=Doe&key=%3CTrelloApiKey%3E&token=%3CAppToken%3E"`;
